---
title: "Ageing impact in gene expression on skeletal muscle repair"
subtitle: "Bioinformatic analysis : Exploration enrichment pathways"
author: "Team Le Grand"
date: "`r format(Sys.time(),'%d %B,%Y')`"
output:
  rmarkdown::html_document:
    theme: lumen
    df_print : paged
    fig_width: 15
    fig_height: 12
    toc: true
    toc_float: true

   
---

<style type="text/css">


h1.title {
  background-color: #158cba;
  font-size: 34px;
  color: white;
}
h1 { /* Header 1 */
  background-color: #158cba;
  font-size: 23px;
    color: white;
}
h2 { /* Header 2 */
  background-color: #158cba ;
  font-size: 20px;
    color: white;
}
h3 { /* Header 3 */
  background-color: #158cba ;
  font-size: 14px;
    color: white;
}
tr:nth-child(even){background-color: #f2f2f2;}

tr:hover {background-color: #ddd;}

th {
  background-color: #84c1d8ff;
  color: white;
}

</style>

# I. Introduction
## I.1 Data origine (Mus musculus - tibialis anterior muscle)
<p>In order to explore cell expression during muscle regeneration in young(2-3 months) and old(18-22 months) cells, RNAseq experiments were performed using cells sorted and FACs at four time points: </p>

  - D0 muscles without injury
  - D2, 2 days after cardiotoxin-induced injury
  - D4, 4 days after injury
  - D7, 7 days after injury

The cell types sampled were as follows:

  - D0 , Endothelial Cells ECs, FibroAdipogenic Progenitors - FAPs, Muscle Stem Cells MuSCs.
  - D2 , ECs, FAPs, MuSCs., Neutrophile Cells, Inflammatory Macrophages - Inflammatory-Mac and Resolving Macrophages - Resolving-Mac .
  - D4, ECs, FAPs, MuSCs, Inflammatory-Mac, Resolving-Mac
  - D7, ECs, FAPs, MuSCs, Resolving-Mac

## I.2 Pretreatment RNA-seq data with pipeline nfcore/rnaseq (v.1.4.2)

<p>The RNA-seq data is aligned to the GRCm38 genome with STAR and counted with FeatureCount .</p>

<p>Only transcripts associated with coding proteins are kept. </p>

<p>Differentially expressed genes (DEG) processed with DESeq2_1.30.1 old vs young (reference expression is young one), for each cell type at each time Day_TypeCell_Old vs Day_TypeCell_Young </p>



```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, rows.print=5)
```
```{r pathsLoads , echo=FALSE,warning=FALSE,message=FALSE }

library(tidyverse)
library(ComplexHeatmap)
library(msigdbr)
library(cowplot)
library(openxlsx)
library(arcdiagram)
library(devtools)
library(igraph)
library(DESeq2)
library(ggrepel) #Â for labels to point
library(rmarkdown)
library(shiny)
library(prettydoc)
library(ggnewscale)
library(scales)
library(htmlwidgets)
library(circlepackeR) 
library(igraph)
library(data.tree)
library(gridExtra)
library(kableExtra)
library(formattable)
library(knitr)
library(DT)
library(ggnewscale)
library(reshape2)
library(oce)
library(ggraph)
library(plyr)
library(spaceNtime)
library(RColorBrewer)
library(patchwork)
library(facetious)

#### Path files to load or created
##################################

#Directories
setwd("~/BulkAnalysis_plusNetwork/")
dir<-"/home/bioinfo/BulkAnalysis_plusNetwork/"
odir <- paste0(dir,"/exam_INTER_conditions/static/")

plots<-"GSEA/PlotsGSEA/"
tableGSEA<-"GSEA/TableGSEA/"
NormData <- "data/CountNormalised/"
HierarchieData= "GSEA/HierarchieData/"

#Loads
gseaoutfull2padjfile = paste0(odir,tableGSEA,"GSEA_table_static_full.rds" )
fullGSEAconcatfile<-paste0(odir,tableGSEA,"GSEA_table_static_full.csv" )
gseaout_filtered_full2padjfile = paste0(odir,tableGSEA,"GSEA_table_static_softfilter.rds" )
countnormalisedfile=paste0(NormData,"CountNormalised.txt")
meancountnormalisedfile=paste0(NormData,"MeanCountNormalised.txt")
DEGtableTotalUPDOWNfile<-paste0(odir,"TableDEG/TableDynamicUpDownDEG.rds")
DEGtablefile<-paste0(odir,"TableDEG/DEG_table_static_full.rds")


#Created
#Completed result GSEA with pattern associated to pathway 
TableGSEADynamics<-"TableGSEAsigniDynamics"
plotpaways<-paste0(plots,"Pathways/")
plotpathwaysHierarchie<-paste0(plots,"PathwaysHierarchie/")
listHierarchieREACTOME<-paste0(odir,HierarchieData,"ListReactomeHiearachieUseFunction.txt")
tablePathwayHierarchieFull<-paste0(odir,HierarchieData,"TableReactomeHiearachieUseFunction.txt")
tablePathwayHierarchieTotalPathwayEnrichie<-paste0(odir,HierarchieData,"TableReactomeHiearachieTotalPathwayEnrichie.txt")

# Report
#Existing
reportDEGanalysis<-"./GeneDEsigniOnMoreOneTimePoint.html"
reportDEGtable<-"./tableDEGsignificantUpDown.html"
reportsuppPathway<-"reportPathwaysEnrichedStatics/"
reportsuppHierarchie<-"reportPathwaysHierarchie/"

##################################
#### Load files
##################################
#rerun code to created y=table and plot

fullGSEA = readRDS(gseaoutfull2padjfile)
fullGSEAconcat<-read.csv(fullGSEAconcatfile,sep = " " )
filteredGSEA =readRDS(gseaout_filtered_full2padjfile) 
countnormalised=read.table(countnormalisedfile,header = T)
meancountnormalised=read.table(meancountnormalisedfile,header = T)
DEGtableTotalUPDOWN<-readRDS(DEGtableTotalUPDOWNfile)
DEGtable<-readRDS(DEGtablefile)

GSEAsigni<-fullGSEAconcat %>% filter(padj<=0.05) %>% mutate(sens=ifelse(NES >0,"Up","Down"))

#Extract patway and gene associated , same use for fgsea analysis
thegmt <- msigdbr(species = "Mus musculus", 
                  category = 'C2', 
                  subcategory=c('CP:REACTOME'))
msigdbr_list = split(x = thegmt$gene_symbol, f = thegmt$gs_name)

genes_df <- read.table("data/genesinfo.csv", sep="\t", header=T)

daysv<-list()
daysv<-lapply( unique(fullGSEAconcat$type), function(t) unique(fullGSEAconcat[fullGSEAconcat$type == t,]$day) )
names(daysv)<-unique(fullGSEAconcat$type)
Typecellv<-unique(fullGSEAconcat$type)

days<-unique(fullGSEAconcat$day)

####################
####Function colors
####################

orderTypecell=c("ECs","FAPs","MuSCs","Neutrophils","Inflammatory-Mac","Resolving-Mac")
colorsType=c("#10b387ff","#3d85c6ff","#b171f1ff","#f0e442ff","#ff9900ff","#cc0000ff")
hsl_color<-data.frame("ECs"=164,"FAPs"=208,"MuSCs"=270,"Neutrophils"=56,"Inflammatory-Mac"=36,"Resolving-Mac"=0)
names(colorsType)=orderTypecell
colorsTime = c(brewer.pal(9,"BuPu")[3],brewer.pal(9,"BuPu")[5],brewer.pal(9,"BuPu")[7],brewer.pal(9,"BuPu")[9])
names(colorsTime)<-days
colorsAge = c("#ffc35d","#019190")
names(colorsAge) = c("Young","Old")
orderDayAge=lapply(days, function(d) c(paste(d,"Young"),paste(d,"Old"))) %>% unlist()

## NewTable with NbCondSigni, CondSigni: Day1_Type1_Sens1.Day_Type2_Sens2 ..., DayTypeSens,NbPathWithDynamique,color, links to report paway
DynamicPathwayAcrossCond <- function(RepartitionPathway,GSEAsigni,n){
  PathwaySigniCond<-RepartitionPathway %>% filter(Freq==n)
  PathwaySigniCondtable<- lapply(PathwaySigniCond$Var1 , function(x) dplyr::filter(GSEAsigni, pathway == x))
  DynamicCond <-lapply(PathwaySigniCondtable, function(x) paste0(x$day,'_',x$type,'_',x$sens) %>% sort())
  names(DynamicCond)<-lapply(PathwaySigniCondtable, function(x) unique(x$pathway) )
  DynamicCondconcat<-c()
  for(Dym in DynamicCond){DynamicCondconcat=c(DynamicCondconcat,str_c(Dym,collapse = " "))}
  names(DynamicCondconcat)<-lapply(PathwaySigniCondtable, function(x) unique(x$pathway))
  Dynamiquetable<-ldply(lapply(PathwaySigniCondtable, function(x) as.data.frame(cbind(rep(n,n),x,paste0(x$day,'_',x$type,'_',x$sens),rep(DynamicCondconcat[[unique(x$pathway)]],n)))),rbind)
  colnames(Dynamiquetable)<-c("NbCondSigni",colnames(PathwaySigniCondtable[[1]]),"DayTypeSens","CondSigni")
  
  Dynamique<-lapply(unique(Dynamiquetable$pathway), function(x) unique(Dynamiquetable %>% filter(pathway==x) %>% select(CondSigni) %>% unlist())) %>% unlist()
  names(Dynamique)<-unique(Dynamiquetable$pathway)
  tableDynamique<-table(Dynamique)
  Dynamiquetable <- Dynamiquetable %>% mutate(NbPathWithDynamique=as.numeric(as.character(tableDynamique[match(Dynamiquetable$CondSigni,names(tableDynamique))]))) %>% mutate(color=lapply(Dynamiquetable$type, function(t) colorsType[[t]]) %>% unlist()) %>% arrange(NbPathWithDynamique,decreasing =T)
  Dynamiquetable <- Dynamiquetable %>% mutate( DayType = paste0(day,'_',type))
  Dynamiquetable <-  Dynamiquetable %>% mutate( link=paste0("<a href='./",reportsuppPathway,"PathwayEnrichedIn",n,"DayTypeCell_",pathway,"_meanCountNormalizedGene.html","'>FigureSupp</a>"))
  Dynamiquetable <-  Dynamiquetable %>% mutate( NamePathway= str_replace(str_replace_all(pathway,"_"," "), "REACTOME","")) %>% mutate( NES = round(Dynamiquetable$NES, digits = 4)) %>% mutate( padj = formatC(Dynamiquetable$padj,format = "e", digits = 4))
  
  return(Dynamiquetable)
}

#Plot of Pathway in function NES
GetPlot<-function(Dynamic6table,GSEAsigni, nbCond , vertical){
  Dynamic6table <-Dynamic6table %>% mutate( CondSigni = as.factor(CondSigni)) %>% mutate(CondSigni = fct_reorder(CondSigni, as.numeric(NbPathWithDynamique), .desc = T) )
  Dynamic6table<-Dynamic6table %>% mutate(type = factor(Dynamic6table$type,orderTypecell ))
  col<- Dynamic6table %>% arrange(type) %>% select(color) %>% unlist() %>% as.character() %>% unique()
  
  typebytime<-ggplot(Dynamic6table, aes(x=day,y=as.numeric(NES),fill=type,color=type,group=paste0(pathway,'_',type))) +
    geom_point(aes(size=as.numeric(NbPathWithDynamique),alpha=0.7),show.legend = c(size=T,alpha=F,col=F,fill=F))+
    scale_size_binned(name="Number path",limits=c(1,MaxNbPaathWithSameDyn))+
    ylab("NES")+
    geom_line()+
    geom_hline(yintercept = 0, color="black", linetype="dashed",lwd=0.5,slope=F)+
    scale_color_manual(name="Cell type",values=col)+
    ggtitle( paste0("Pathways with ",nbCond," conditions enriched")) +
    facet_wrap(~CondSigni+NbPathWithDynamique ,dir = "v", ncol=vertical,labeller = labeller(NbPathWithDynamique = label_facet(Dynamic6table$NbPathWithDynamique,"Number pathways: ")))+
    theme_linedraw()+ theme(strip.text  = element_text(size = 11 , face = "bold"),strip.background = element_rect(fill="grey"),panel.grid.major.x = element_line(color = "gray60", size = 0.8),axis.title= element_text(size=14,face="bold"),axis.text = element_text(face="bold",size=12),legend.position="bottom",legend.title = element_text(size=14,face="bold"),legend.text = element_text(size=12,face="bold"))
  print(typebytime)
  save_plot(paste0(odir, plots, 'Pathway_enriched_in_',nbCond ,'conditions_Plot2.svg'),typebytime, base_height = 10,
            base_width = 14)
}


```

## I.3 Differential Expression Genes Analysis

Report [linksReportDEGanalysis](`r reportDEGanalysis`)

Report [linksTableDEG](`r reportDEGtable`)

# II Genes Set Enrichment Analysis (GSEA) results

--------

GSEA on all genes. This analysis is done by day and cell type on list of genes ranked by logfoldchange sign * padj decreasing. The data base use is genes set from Molecular Signatures DataBase (MSigDB) species : 

  - species  = Mus musculus 
  - category C2 = online pathway databases and the biomedical literature curated ([more details](https://www.gsea-msigdb.org/gsea/msigdb/collection_details.jsp#C2) )
  - subcategory = Canonical pathways ,Reactome database
Composed by 1604 pathways. 


## II.1 Descriptives statistics

--------

How many pathways are significants (padj<0.05)

```{r caclcStat1, echo=FALSE,warning=FALSE,message=FALSE }
################################################# 
# Extract stats significant pathways enrichied
#################################################

NumberPathwaysSigni = array(NA, dim=c(length(days),length(Typecellv)))
rownames(NumberPathwaysSigni) =  days
colnames(NumberPathwaysSigni) = Typecellv
NumberPathwaysUniqueOnOneDay<-c()
NumberPathwaysUniqueOnOneTypeCell<-c()
for (t in Typecellv){
  link=paste0("[", GSEAsigni %>% filter(type==t) %>% select(pathway) %>% unique() %>% unlist() %>% length(),"](./",reportsuppHierarchie,"/",t,"_Hierarchical_Figsupp.html)")
  NumberPathwaysUniqueOnOneTypeCell<-c(NumberPathwaysUniqueOnOneTypeCell,link)
  for (d in daysv[[t]]){
    link=paste0("[",dim(GSEAsigni %>% filter(day==d & type==t))[1],"](./",reportsuppHierarchie,"/",d,'_',t,"_Hierarchical_Figsupp.html)")
    if ( dim(GSEAsigni %>% filter(day==d & type==t))[1] == 0){
      NumberPathwaysSigni[d, t] <-0
      }
    else{NumberPathwaysSigni[d, t] <- link}
    if (t == Typecellv[1]){
      link=paste0("[",GSEAsigni %>% filter(day==d) %>% select(pathway) %>% unique() %>% unlist() %>% length(),"](","./",reportsuppHierarchie,"/",d,"_Hierarchical_Figsupp.html)")
       NumberPathwaysUniqueOnOneDay<-c(NumberPathwaysUniqueOnOneDay,link)
    }
  }
}

names(NumberPathwaysUniqueOnOneDay)<-days
names(NumberPathwaysUniqueOnOneTypeCell)<-Typecellv
NumberPathwaysSigni<-cbind(NumberPathwaysSigni,NumberPathwayEnrichedByDay=NumberPathwaysUniqueOnOneDay)
NumberPathwaysSigni<-rbind(NumberPathwaysSigni,NumberPathwayEnrichedByTypeCell=c(NumberPathwaysUniqueOnOneTypeCell,""))
TotalUniquePathway=length(unique(GSEAsigni$pathway))

```


`r kbl(NumberPathwaysSigni,row.names = TRUE)%>%kable_paper("striped")`



```{r violinPlotAllNES, fig.align='center', echo=FALSE, warning=FALSE , error=FALSE}
###  CALCULATE
# Visualization of path significative
#
# ====================== Calculate =============================
NumberPathwaysSigni = array(NA, dim=c(length(days),length(Typecellv)))
rownames(NumberPathwaysSigni) =  days
colnames(NumberPathwaysSigni) = Typecellv

NumberPathwaysUniqueOnOneDay<-c()
NumberPathwaysUniqueOnOneTypeCell<-c()
for (d in days){
  NumberPathwaysUniqueOnOneDay<-c(NumberPathwaysUniqueOnOneDay,GSEAsigni %>% filter(day==d) %>% select(pathway) %>% unique() %>% unlist() %>% length())
  for (t in Typecellv){
    NumberPathwaysSigni[d, t] <- dim(GSEAsigni %>% filter(day==d & type==t))[1]
    if (d == days[1]){
      NumberPathwaysUniqueOnOneTypeCell<-c(NumberPathwaysUniqueOnOneTypeCell, GSEAsigni %>% filter(type==t) %>% select(pathway) %>% unique() %>% unlist() %>% length())
    }
  }
}

names(NumberPathwaysUniqueOnOneDay)<-days
names(NumberPathwaysUniqueOnOneTypeCell)<-Typecellv
Violindata<-data.frame(ViolinDT=paste0(GSEAsigni$day,"_",GSEAsigni$type),
                       ViolinDay=factor(GSEAsigni$day,levels= unique(GSEAsigni$day)),
                       CellType=factor(GSEAsigni$type, levels= orderTypecell),
                       ViolinValue=GSEAsigni$NES,
                       nbPathwaySignificant = lapply(1:length(GSEAsigni$NES), function(i) NumberPathwaysSigni[GSEAsigni[i,]$day,GSEAsigni[i,]$type] ) %>% unlist(),
                       a=rep("a",length(GSEAsigni$NES)))


PlotDEGonlyOneDayVioline<-ggplot(Violindata, aes(x=a ,y=ViolinValue, fill=nbPathwaySignificant)) +# fill=name allow to automatically dedicate a color for each group
  geom_jitter( shape=16, size=3, position=position_jitter(0.4), aes(color=nbPathwaySignificant,alpha=0.5), show.legend=F)+
  geom_violin(scale="count",aes(alpha=0.01),show.legend = c("alpha"=F))+
  scale_fill_gradient(low = "#E7E1EF",high="#67001F",labels = scales::label_comma())+scale_color_gradient(low = "#E7E1EF",high="#67001F",labels = scales::label_comma())+
  coord_flip()+
  scale_x_discrete(limits=rev)  +   facet_grid_blank(vars(CellType),vars(ViolinDay), drop = FALSE) +
  geom_hline(yintercept = 0, color="black", linetype="dashed",lwd=0.5)+
  ylab("NES")+xlab("Type Cell") +
  theme(axis.title = element_text(size=12, face = "bold"),axis.text.x=element_text(size=7,colour = "black", face="bold"),axis.text.y=element_text(colour = "white"),legend.title=element_text( size=12, face = "bold"),legend.text=element_text(size=5),title=element_text(size=14, face = "bold"),
        legend.spacing= unit(0.1,"points"),legend.position = "bottom",legend.spacing.y=unit(0.01,"mm"),legend.key.size = unit(3, "mm"),panel.grid =element_line(color="white"),strip.text.x = element_text(color = "white", face= "bold",size=14),strip.text.y = element_text(color = "black",size=12),
        panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_blank())+labs(fill = "Count pathway enrichied padj<0.05")+
  ggtitle("Distribution of significant pathway enriched  Old vs Young \n according to NES")
PlotDEGonlyOneDayVioline
#Add color in background grid
g5 <- ggplot_gtable(ggplot_build(PlotDEGonlyOneDayVioline))
stripRowName <- which(grepl('strip', g5$layout$name))
k <- 1
fills <- c(colorsTime,colorsType)
for (i in stripRowName) {
  j <- which(grepl('rect', g5$grobs[[i]]$grobs[[1]]$childrenOrder))
  g5$grobs[[i]]$grobs[[1]]$children[[j]]$gp$fill <- fills[k]
  k <- k+1
}
grid::grid.draw(g5)
```

## II.2 Inspected Hierarchical Classement of pathway in REACTOME

--------

All condition combined, there are `r TotalUniquePathway` pathway enriched. 
In this representation only pathway with upper letter are enriched in at least one condition.

-----------

```{r HierarchicalTotalPlot, echo=FALSE,warning=FALSE,message=FALSE }
data_nested_2=read.table( file=paste0( odir,HierarchieData,"ALL/ALL_table_REACTOME_Hierachie.txt"))
data_nested_2[is.na(data_nested_2)] <- ""
data_nested_2$pathString <- paste("roots", data_nested_2$level1, data_nested_2$level2, data_nested_2$level3, data_nested_2$level4,data_nested_2$level5,data_nested_2$level6,data_nested_2$level7, data_nested_2$level8, data_nested_2$level9, data_nested_2$level10,data_nested_2$level11,data_nested_2$level12, sep = "/")
data_Node <-as.Node(data_nested_2)

p <- circlepackeR(data_Node, size = "value",color_min = paste0("hsl(56,80%,80%)"), color_max = paste0("hsl(341,30%,30%)"), width = 1800, height = 1000)

p

```

----------`

```{r HierarchicalTotaltable, echo=FALSE,warning=FALSE,message=FALSE }

tableHierarchie<-read.table(paste0(odir,HierarchieData,"ALL_table_REACTOME_Hierachie_print.txt") )
 DT::datatable(tableHierarchie, filter = 'top', extensions = c('RowGroup','Select', 'SearchPanes'),options = list(rowGroup = list(dataSrc = c(1,2)),dom = 'Pfrtip',columnDefs = list(list(searchPanes = list(show = FALSE), targets = 3:12 ))),selection = 'none',escape = F)
```


--------

## II.3 Inspected the repartition of enrichied pathways across days and type cell 

------------

```{r caclcStat2, echo=FALSE,warning=FALSE,message=FALSE }
RepartitionPathwaySummary<-table(table(GSEAsigni$pathway))
RepartitionPathwaySummary<- data.frame("Numbers conditions enrichis"="Numbers pathways","1"=RepartitionPathwaySummary[1], "2"=RepartitionPathwaySummary[2],"3"=RepartitionPathwaySummary[3],"4"=RepartitionPathwaySummary[4],"5"=RepartitionPathwaySummary[5],"6"=RepartitionPathwaySummary[6],"7"=RepartitionPathwaySummary[7],"8"=RepartitionPathwaySummary[8])
colnames(RepartitionPathwaySummary)=c("Numbers conditions enrichis",1:8)
RepartitionPathway<-as.data.frame(table(GSEAsigni$pathway))


```

Number pathway `r kbl(RepartitionPathwaySummary,row.names = F)%>%kable_paper("striped")`


```{r CalTableDynamic, echo=FALSE,warning=FALSE,message=FALSE,  paged.print=TRUE, fig.show='asis', fig.fullwidth=TRUE}
RepartitionPathwaySummary<-table(table(GSEAsigni$pathway))
## dataframe : Var1 = pathways Freq = Number condition which the pathway is found
RepartitionPathway<-as.data.frame(table(GSEAsigni$pathway)) 

Dynamic8table=DynamicPathwayAcrossCond(RepartitionPathway,GSEAsigni,8)
Dynamic7table=DynamicPathwayAcrossCond(RepartitionPathway,GSEAsigni,7)
Dynamic6table=DynamicPathwayAcrossCond(RepartitionPathway,GSEAsigni,6)
Dynamic5table=DynamicPathwayAcrossCond(RepartitionPathway,GSEAsigni,5)
Dynamic4table=DynamicPathwayAcrossCond(RepartitionPathway,GSEAsigni,4)
Dynamic3table=DynamicPathwayAcrossCond(RepartitionPathway,GSEAsigni,3)
Dynamic2table=DynamicPathwayAcrossCond(RepartitionPathway,GSEAsigni,2)
Dynamic1table=DynamicPathwayAcrossCond(RepartitionPathway,GSEAsigni,1)

MaxNbPaathWithSameDyn<-max(Dynamic8table$NbPathWithDynamique,Dynamic7table$NbPathWithDynamique,Dynamic6table$NbPathWithDynamique,Dynamic5table$NbPathWithDynamique,Dynamic4table$NbPathWithDynamique,Dynamic3table$NbPathWithDynamique,Dynamic2table$NbPathWithDynamique,Dynamic1table$NbPathWithDynamique )

```

### II.3.1 Pathways founds enriched in 8 conditions

-------

```{r PlotcombiPathwayCond8, echo=FALSE,warning=FALSE,message=FALSE,  paged.print=TRUE, fig.show='asis', fig.fullwidth=TRUE}
GetPlot(Dynamic8table,GSEAconcat,8,1)
listPlotTable <- Dynamic8table %>% select(CondSigni,NamePathway,link,DayTypeSens,padj,NES,NbPathWithDynamique)
DT::datatable(listPlotTable, filter = 'top', extensions = c('RowGroup','Select', 'SearchPanes'),options = list(rowGroup = list(dataSrc = c(1,2)),dom = 'Pfrtip',columnDefs = list(list(searchPanes = list(show = FALSE), targets = 3:7 ))),selection = 'none',escape = F)
```

### II.3.2 Pathways founds enriched in 7 conditions

----------

```{r PlotcombiPathwayCond7, echo=FALSE,warning=FALSE,message=FALSE,  paged.print=TRUE, fig.show='asis', fig.fullwidth=TRUE}
GetPlot(Dynamic7table,GSEAconcat,7,1)
listPlotTable <- Dynamic7table %>% select(CondSigni,NamePathway,link,DayTypeSens,padj,NES,NbPathWithDynamique)
DT::datatable(listPlotTable, filter = 'top', extensions = c('RowGroup','Select', 'SearchPanes'),options = list(rowGroup = list(dataSrc = c(1,2)),dom = 'Pfrtip',columnDefs = list(list(searchPanes = list(show = FALSE), targets = 3:7 ))),selection = 'none',escape = F)
```


### II.3.3 Pathways founds enriched in 6 conditions

-------

```{r PlotcombiPathwayCond6, echo=FALSE,warning=FALSE,message=FALSE,  paged.print=TRUE, fig.show='asis', fig.fullwidth=TRUE}
GetPlot(Dynamic6table,GSEAconcat,6,1)
listPlotTable <- Dynamic6table %>% select(CondSigni,NamePathway,link,DayTypeSens,padj,NES,NbPathWithDynamique)
DT::datatable(listPlotTable, filter = 'top', extensions = c('RowGroup','Select', 'SearchPanes'),options = list(rowGroup = list(dataSrc = c(1,2)),dom = 'Pfrtip',columnDefs = list(list(searchPanes = list(show = FALSE), targets = 3:7 ))),selection = 'none',escape = F)
```

### II.3.4 Pathways founds enriched in 5 conditions

------

```{r PlotcombiPathwayCond5, echo=FALSE,warning=FALSE,message=FALSE,  fig.show='asis', paged.print=TRUE}
GetPlot(Dynamic5table,GSEAconcat,5,2)
listPlotTable <- Dynamic5table %>% select(CondSigni,NamePathway,link,DayTypeSens,padj,NES,NbPathWithDynamique)
DT::datatable(listPlotTable, filter = 'top', extensions = c('RowGroup','Select', 'SearchPanes'),options = list(rowGroup = list(dataSrc = c(1,2)),dom = 'Pfrtip',columnDefs = list(list(searchPanes = list(show = FALSE), targets = 3:7 ))),selection = 'none',escape = F)
```

### II.3.5 Pathways founds enriched in 4 conditions

--------

```{r PlotcombiPathwayCond4, echo=FALSE, fig.show='asis', message=FALSE, warning=FALSE, paged.print=TRUE}
GetPlot(Dynamic4table,GSEAconcat,4,2)
listPlotTable <- Dynamic4table %>% select(CondSigni,NamePathway,link,DayTypeSens,padj,NES,NbPathWithDynamique)
DT::datatable(listPlotTable, filter = 'top', extensions = c('RowGroup','Select', 'SearchPanes'),options = list(rowGroup = list(dataSrc = c(1,2)),dom = 'Pfrtip',columnDefs = list(list(searchPanes = list(show = FALSE), targets = 3:7 ))),selection = 'none',escape = F)

``` 

### Pathways founds enriched in 3 conditions

--------

```{r PlotcombiPathwayCond3, echo=FALSE, fig.show='asis', message=FALSE, warning=FALSE, paged.print=TRUE}
GetPlot(Dynamic3table,GSEAconcat,3,NA)
listPlotTable <- Dynamic3table %>% select(CondSigni,NamePathway,link,DayTypeSens,padj,NES,NbPathWithDynamique)
DT::datatable(listPlotTable, filter = 'top', extensions = c('RowGroup','Select', 'SearchPanes'),options = list(rowGroup = list(dataSrc = c(1,2)),dom = 'Pfrtip',columnDefs = list(list(searchPanes = list(show = FALSE), targets = 3:7 ))),selection = 'none',escape = F)

``` 

### II.3.6 Pathways founds enriched in 2 conditions

---------

```{r PlotcombiPathwayCond2, echo=FALSE, fig.show='asis', message=FALSE, warning=FALSE, paged.print=TRUE}
GetPlot(Dynamic2table,GSEAconcat,2,NA)
listPlotTable <- Dynamic2table %>% select(CondSigni,NamePathway,link,DayTypeSens,padj,NES,NbPathWithDynamique)
DT::datatable(listPlotTable, filter = 'top', extensions = c('RowGroup','Select', 'SearchPanes'),options = list(rowGroup = list(dataSrc = c(1,2)),dom = 'Pfrtip',columnDefs = list(list(searchPanes = list(show = FALSE), targets = 3:7 ))),selection = 'none',escape = F)
``` 

### II.3.7 Pathways founds enrichired in 1 condition

-----


```{r PlotcombiPathwayCond1, echo=FALSE, fig.show='asis', message=FALSE, warning=FALSE, paged.print=TRUE}
GetPlot(Dynamic1table,GSEAconcat,1,NA)
listPlotTable <- Dynamic1table %>% select(CondSigni,NamePathway,link,DayTypeSens,padj,NES,NbPathWithDynamique)
DT::datatable(listPlotTable, filter = 'top', extensions = c('RowGroup','Select', 'SearchPanes'),options = list(rowGroup = list(dataSrc = c(1)),dom = 'Pfrtip',columnDefs = list(list(searchPanes = list(show = FALSE), targets = 3:7 ))),selection = 'none',escape = F)
``` 

### Table Recapitulatif

-----


```{r TableRecap, echo=FALSE, fig.show='asis', message=FALSE, warning=FALSE, paged.print=TRUE}
Dynamictable<-rbind(Dynamic8table,Dynamic7table,Dynamic6table,Dynamic5table,Dynamic4table,Dynamic3table,Dynamic2table,Dynamic1table)
listPlotTable <- Dynamictable %>% select(DayTypeSens,NamePathway,CondSigni,NbCondSigni,link,day,type,padj,NES,NbPathWithDynamique) %>% arrange(DayTypeSens,NamePathway,CondSigni,desc(NbCondSigni))
DT::datatable(listPlotTable, filter = 'top', extensions = c('RowGroup','Select', 'SearchPanes'),options = list(rowGroup = list(dataSrc = c(1,2,3)),dom = 'Pfrtip',columnDefs = list(list(searchPanes = list(show = FALSE), targets = 4:9 ))),selection = 'none',escape = F)
``` 
