---
title: "Ageing impact in gene expression on skeletal muscle repair"
subtitle: "Bioinformatic analysis : differential expression genes part"
author: "Team Le Grand"
date: "`r format(Sys.time(),'%d %B,%Y')`"
output:
  rmarkdown::html_document:
    theme: lumen
    df_print : paged
    fig_width: 10
    fig_height: 8
    toc: true
    toc_float: true
   
---

<style type="text/css">


h1.title {
  background-color: #158cba;
  font-size: 34px;
  color: white;
}
h1 { /* Header 1 */
  background-color: #158cba;
  font-size: 23px;
    color: white;
}
h2 { /* Header 2 */
  background-color: #158cba ;
  font-size: 20px;
    color: white;
}
h3 { /* Header 3 */
  background-color: #158cba ;
  font-size: 14px;
    color: white;
}
tr:nth-child(even){background-color: #f2f2f2;}

tr:hover {background-color: #ddd;}

th {
  background-color: #84c1d8ff;
  color: white;
}

.btn {
    border-width: 0 0px 0px 0px;
    font-weight: normal;
    text-transform: ;
}
.btn-default {
    color: #2ecc71;
    background-color: #ffffff;
    border-color: #ffffff;
}

</style>

# I.Data origine (Mus musculus - tibialis anterior muscle)
<p>In order to explore cell expression during muscle regeneration in young(2-3 months) and old(18-22 months) cells, RNAseq experiments were performed using cells sorted and FACs at four time points: </p>

  - D0 muscles without injury
  - D2, 2 days after cardiotoxin-induced injury
  - D4, 4 days after injury
  - D7, 7 days after injury

The cell types sampled were as follows:

  - D0 , Endothelial Cells ECs, FibroAdipogenic Progenitors - FAPs, Muscle Stem Cells MuSCs.
  - D2 , ECs, FAPs, MuSCs., Neutrophile Cells, Inflammatory Macrophages - Inflammatory-Mac and Resolving Macrophages - Resolving-Mac .
  - D4, ECs, FAPs, MuSCs, Inflammatory-Mac, Resolving-Mac
  - D7, ECs, FAPs, MuSCs, Resolving-Mac

## I.1 Pretreatment RNA-seq data with pipeline nfcore/rnaseq (v.1.4.2)
<p>The RNA-seq data is aligned to the GRCm38 genome with STAR and counted with FeatureCount .</p>

<p>Only transcripts associated with coding proteins are kept. </p>

<p>Differentially expressed genes (DEG) processed with DESeq2_1.30.1 old vs young (reference expression is young one), for each cell type at each time Day_TypeCell_Old vs Day_TypeCell_Young </p>

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)

```

```{r pathsLoads , echo=FALSE,warning=FALSE,message=FALSE }

library(tidyverse)
library(ComplexHeatmap)
library(msigdbr)
library(cowplot)
library(treemap)
library(networkD3)
library(dplyr)
library(ggplot2)
library(hrbrthemes)
library(webshot)
library(htmlwidgets)
library(gridExtra)
library(kableExtra)
library(RColorBrewer)
library(facetious)
library(formattable)
library(grid)
library(PantaRhei)
library(DT)
library(reshape2)
#setwd("~/BulkAnalysis_plusNetwork/")

#Directories
odir <- "/home/bioinfo/BulkAnalysis_plusNetwork/exam_INTER_conditions/static/"
DEG_table = "TableDEG/DEG_table_"
plot<-'PlotsDEG/'
NormData <- "/home/bioinfo/BulkAnalysis_plusNetwork/data/CountNormalised/"

#To created
analyse<-paste0(odir,plot,"DynamiqueDEGbytypeCell/")
if (dir.exists(analyse) == F) {
  dir.create(analyse)
}
pathcombinaisontable<-paste0(odir,"TableDEG/TableDynamicDEG")
pathcombinaisontableUPDOWN<-paste0(odir,"TableDEG/TableDynamicUpDownDEG")

#Loading
#DEG table
fullDEsta = readRDS(paste0(odir,DEG_table, "static_full.rds"))
#Gene ID vs symbol
genes_df <- read.table("/home/bioinfo/BulkAnalysis_plusNetwork/data/genesinfo.csv", sep="\t", header=T)

#Get day and type cell
daysv<-list()
daysv<-lapply( unique(fullDEsta$type), function(t) unique(fullDEsta[fullDEsta$type == t,]$day) )
names(daysv)<-unique(fullDEsta$type)
Typecellv<-unique(fullDEsta$type)
days<-unique(fullDEsta$day)
vectTypeCell <-unique(fullDEsta$type)

#Plot order + color day and type cell
orderTypecell=c("ECs","FAPs","MuSCs","Neutrophils","Inflammatory-Mac","Resolving-Mac")
colorsType=c("#10b387ff","#3d85c6ff","#b171f1ff","#f0e442ff","#ff9900ff","#cc0000ff")
names(colorsType)=orderTypecell
colorsTime = c(brewer.pal(9,"BuPu")[3],brewer.pal(9,"BuPu")[5],brewer.pal(9,"BuPu")[7],brewer.pal(9,"BuPu")[9])
names(colorsTime)<-days

```


```{r caclcStat1, echo=FALSE,warning=FALSE,message=FALSE }

TotUniqueGene<-length(unique(fullDEsta$id))
signiDEgene <- fullDEsta %>% filter(padj<=0.05)
TotUniqueGeneDE<-length(unique(signiDEgene$id))
TotGeneDED0<-length(signiDEgene[signiDEgene$day == 'D0',]$id)
TotUniqueGeneDED0<-length(unique(signiDEgene[signiDEgene$day == 'D0',]$id))

padjofpathsDEsigni = array(NA, dim=c(4,6))
rownames(padjofpathsDEsigni) = unique(fullDEsta$day)
colnames(padjofpathsDEsigni) = unique(fullDEsta$type)

for (d in unique(fullDEsta$day)){
  for (n in unique(fullDEsta$type)){
    padjofpathsDEsigni[d, n] <- round(length(fullDEsta %>% filter(day == d & type == n & padj <=0.05) %>% dplyr::select(symbol)  %>% unlist() %>% unique())/TotUniqueGeneDE*100, digits = 1)
    if ( padjofpathsDEsigni[d, n] == 0 ){padjofpathsDEsigni[d, n]=NA}
  }
}

totalDay = lapply( days , function(d) round(length(signiDEgene %>% filter(day==d) %>% select(symbol) %>% unlist() %>% unique())/TotUniqueGeneDE*100, digits = 1)) %>% unlist()
toltalTypeCell = lapply( Typecellv , function(t) round(length(signiDEgene %>% filter(type==t) %>% select(symbol) %>% unlist() %>% unique())/TotUniqueGeneDE*100, digits = 1)) %>% unlist()
totalDay = c(totalDay,"")

for(c in 1:dim(padjofpathsDEsigni)[2]){ padjofpathsDEsigni[,c]<-ifelse(as.numeric(padjofpathsDEsigni[,c]) == max(as.numeric(padjofpathsDEsigni[,c]),na.rm = T),cell_spec(padjofpathsDEsigni[,c], color = colorsType[colnames(padjofpathsDEsigni)][[c]] , bold = T), cell_spec(padjofpathsDEsigni[,c],color ="black"))}
padjofpathsDEsigni = rbind(padjofpathsDEsigni, "PercentDEGSigniOnTyCell"= toltalTypeCell)
padjofpathsDEsigni = cbind(padjofpathsDEsigni, "PercentDEGSigniOnDay"= totalDay)

padjofpathsDEsigni<-as.data.frame(padjofpathsDEsigni)
padjofpathsDEsigni$PercentDEGSigniOnDay<-ifelse(as.numeric(padjofpathsDEsigni$PercentDEGSigniOnDay) == max(as.numeric(padjofpathsDEsigni$PercentDEGSigniOnDay),na.rm = T),cell_spec(padjofpathsDEsigni$PercentDEGSigniOnDay,bold = T), cell_spec(padjofpathsDEsigni$PercentDEGSigniOnDay,color ="black"))
padjofpathsDEsigni<-t(padjofpathsDEsigni)
padjofpathsDEsigni<-as.data.frame(padjofpathsDEsigni)
padjofpathsDEsigni$PercentDEGSigniOnTyCell<-ifelse(as.numeric(padjofpathsDEsigni$PercentDEGSigniOnTyCell) == max(as.numeric(padjofpathsDEsigni$PercentDEGSigniOnTyCell),na.rm = T),cell_spec(padjofpathsDEsigni$PercentDEGSigniOnTyCell,bold = T), cell_spec(padjofpathsDEsigni$PercentDEGSigniOnTyCell,color ="black"))

```
# II.Overview of DEG significant analysis {.tabset .tabset-fade .tabset-pills}
***

On any given day and cell type there are `r TotUniqueGeneDE` genes differentilly expressed with a good confidence (padj<=0.05) (= DEG significative) on `r TotUniqueGene` genes expressed in total or **`r round(TotUniqueGeneDE/TotUniqueGene*100, digits = 0)`%** of DEG significant.

The table below, is the percent of DEG significant on total genes expressed.
## Percent DEG significant table
`r kbl(padjofpathsDEsigni,row.names = TRUE,escape=F)%>%kable_paper("striped")`

## Number DEG significant table



```{r ViolinePlotGlobal, fig.align='center', echo=FALSE, warning=FALSE , error=FALSE}
###  CALCULATE
# Found the distribution of genes significatifs on Days Conditions by type cell
# Visualization for genes DE on only 1 days 
#
# ====================== Calculate =============================
signiDEgene <- fullDEsta %>% filter(padj<=0.05)

###  CALCULATE
# Distribution general of genes DE log2foldchange
signiDEgene<-signiDEgene %>% mutate(day.type=paste0(day,'_',type))
nbGeneSignificant<-table(signiDEgene$day.type)
#dataframe for violin plot
Violindata<-data.frame(ViolinDT=paste0(signiDEgene$day,"_",signiDEgene$type),
                       ViolinDay=factor(signiDEgene$day,levels= unique(signiDEgene$day)),
                       CellType=factor(signiDEgene$type, levels= orderTypecell),
                       ViolinValue=signiDEgene$log2FoldChange,
                       nbGeneSignificant = lapply(signiDEgene$day.type, function(dt) nbGeneSignificant[[dt]]) %>% unlist(),
                       a=rep("a",length(signiDEgene$day.type)))

PlotDEGonlyOneDayVioline<-ggplot(Violindata, aes(x=a ,y=ViolinValue, fill=nbGeneSignificant)) +# fill=name allow to automatically dedicate a color for each group
  geom_jitter( shape=16, size=1, position=position_jitter(0.4), aes(color=nbGeneSignificant,alpha=0.001), show.legend=F)+
  geom_violin(scale="count",aes(alpha=0.001),show.legend = c("alpha"=F))+
  scale_fill_gradient(low = "#E7E1EF",high="#67001F",labels = scales::label_comma())+scale_color_gradient(low = "#E7E1EF",high="#67001F",labels = scales::label_comma())+
  coord_flip()+
  scale_x_discrete(limits=rev)  +   facet_grid_blank(vars(CellType),vars(ViolinDay), drop = FALSE) +
  geom_hline(yintercept = 0, color="black", linetype="dashed",lwd=0.5)+
  ylab("log2FoldChange")+xlab("Type Cell") +
  theme(axis.title = element_text(size=12, face = "bold"),axis.text.x=element_text(size=7,colour = "black", face="bold"),axis.text.y=element_text(colour = "white"),legend.title=element_text( size=12, face = "bold"),legend.text=element_text(size=4),title=element_text(size=14, face = "bold"),
        legend.spacing= unit(0.1,"points"),legend.position = "bottom",legend.spacing.y=unit(0.01,"mm"),legend.key.size = unit(3, "mm"),panel.grid =element_line(color="white"),strip.text.x = element_text(color = "white", face= "bold",size=12),strip.text.y = element_text(color = "black", face= "bold",size=10),
        panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_blank())+labs(fill = "Count DEG padj<0.05")+
  ggtitle("Distribution of significatif differentially expressed genes Old vs Young \n according to log2Foldchange")
#Add color in background grid
g5 <- ggplot_gtable(ggplot_build(PlotDEGonlyOneDayVioline))
stripRowName <- which(grepl('strip', g5$layout$name))
k <- 1
fills <- c(colorsTime,colorsType)
for (i in stripRowName) {
  j <- which(grepl('rect', g5$grobs[[i]]$grobs[[1]]$childrenOrder))
  g5$grobs[[i]]$grobs[[1]]$children[[j]]$gp$fill <- fills[k]
  k <- k+1
}
grid::grid.draw(g5)
#png(paste0(odir,plot,"PlotDEGsigniDayTypeVioline.png"),units = "in", width=5, height= 7, res = 300, family = "Arial")
#grid::grid.draw(g5)
#dev.off()

```


Graphics and tables are in **`r odir`**

# III.Exploration of DEG significant patterns in one type cell on muscle repair
This exploration is type cell specific and focus on gene differentially expressed significant (padj<0.05)
On one type Cell, we observe different dynamics of differential expression  across days
  
  - For some gene, the difference between Old and Young are specific of one day, on one step of muscle regeneration.
  - For some gene, the difference is always present, no specific.
  - For some gene, the age up his expression in some days and down in other day.


```{r caclcStat3, echo=FALSE,warning=FALSE,message=FALSE }
dataFrameTotalUPDOWN<-readRDS(paste0(pathcombinaisontableUPDOWN,'.rds'))

# Number Gene DE on Only one TypeCell and the percent 
TotUniqueGene<-Reduce(`+`,lapply(unique(fullDEsta$type), function(t) filter(fullDEsta,type == t) %>% dplyr::select(id) %>% unlist() %>% as.character() %>% unique() %>% length()))
TotUniqueGeneDE<-Reduce(`+`,lapply(unique(signiDEgene$type), function(t) filter(signiDEgene,type == t) %>% dplyr::select(id) %>% unlist() %>% as.character()%>% unique %>% length()))

dataframeDEGonecond <- lapply(unique(dataFrameTotalUPDOWN$type), function(t) filter(dataFrameTotalUPDOWN,type == t) %>% filter(nbDayDE == 1))
names(dataframeDEGonecond)<-unique(dataFrameTotalUPDOWN$type)
nbUniqGeneDEonecond<-  Reduce(`+`,lapply(dataframeDEGonecond, function(d) unique(d$id) %>% length()))

dataframeDEGtwocond <- lapply(unique(dataFrameTotalUPDOWN$type), function(t) filter(dataFrameTotalUPDOWN,type == t) %>% filter(nbDayDE == 2))
names(dataframeDEGtwocond)<-unique(dataFrameTotalUPDOWN$type)
nbUniqGeneDEtwocond<-  Reduce(`+`,lapply(dataframeDEGtwocond, function(d) unique(d$id) %>% length()))

dataframeDEGthreecond <- lapply(unique(dataFrameTotalUPDOWN$type), function(t) filter(dataFrameTotalUPDOWN,type == t) %>% filter(nbDayDE == 3))
names(dataframeDEGthreecond)<-unique(dataFrameTotalUPDOWN$type)
nbUniqGeneDEthreecond<-  Reduce(`+`,lapply(dataframeDEGthreecond, function(d) unique(d$id) %>% length()))

dataframeDEGfourcond <- lapply(unique(dataFrameTotalUPDOWN$type), function(t) filter(dataFrameTotalUPDOWN,type == t) %>% filter(nbDayDE == 4))
names(dataframeDEGfourcond)<-unique(dataFrameTotalUPDOWN$type)
nbUniqGeneDEfourcond<-  Reduce(`+`,lapply(dataframeDEGfourcond, function(d) unique(d$id) %>% length()))

# Number Gene DE on 2, 3 and 4 TypeCell and the percent
tablePercentgeneDEacrosscond<- data.frame("one Day"= c(nbUniqGeneDEonecond,round(nbUniqGeneDEonecond/TotUniqueGeneDE*100,digits = 0)),
                                          "two Day"=c(nbUniqGeneDEtwocond,round(nbUniqGeneDEtwocond/TotUniqueGeneDE*100,digits = 0)),
                                          "three Day"=c(nbUniqGeneDEthreecond,round(nbUniqGeneDEthreecond/TotUniqueGeneDE*100, digits = 0)),
                                          "four Day"=c(nbUniqGeneDEfourcond, round(nbUniqGeneDEfourcond/TotUniqueGene*100, digits = 0)),
                                          "names"=c("Number Gene DE on ","Pourcent Gene DE on"))
tablePercentgeneDEacrosscond=column_to_rownames(tablePercentgeneDEacrosscond, var = "names")
#In which type cell there are a number max of gene DE in 2  or 3 or 4 days

i=0
for (t in dataframeDEGtwocond){ i=i+1;  if (dim(t)[1] <1){ dataframeDEGtwocond<-dataframeDEGtwocond[-i] ; i=i-1 }}
i=0
for (t in dataframeDEGthreecond){ i=i+1;  if (dim(t)[1] <1){dataframeDEGthreecond<-dataframeDEGthreecond[-i]; i=i-1}}
i=0
for (t in dataframeDEGfourcond){ i=i+1;  if (dim(t)[1] <1){dataframeDEGfourcond<-dataframeDEGfourcond[-i]; i=i-1}}

maxGE1cond1<-lapply(names(dataframeDEGonecond), function(x) dataframeDEGonecond[[x]] %>% dplyr::select(CombiDayDEG) %>% unlist() %>% as.character() %>% table() %>% as.data.frame() %>% top_n(1) %>% mutate("numberGene"=Freq) %>% as.data.frame())
maxGE1cond2<-lapply(names(dataframeDEGonecond), function(x) dataframeDEGonecond[[x]] %>% dplyr::select(CombiDaysUpDown) %>% unlist() %>% as.character() %>% table() %>% as.data.frame() %>% top_n(1) %>% mutate("numberGene"=Freq) %>% as.data.frame())
maxGE1cond<-lapply(1:length(maxGE1cond1), function(i) cbind(rbind(maxGE1cond1[[i]],maxGE1cond2[[i]]),numberDay=rep(1,dim(rbind(maxGE1cond1[[i]],maxGE1cond2[[i]]))[[1]]),typecell=rep(names(dataframeDEGonecond)[[i]],dim(rbind(maxGE1cond1[[i]],maxGE1cond2[[i]]))[[1]])))
names(maxGE1cond) = names(dataframeDEGonecond)


maxGE2cond1<-lapply(names(dataframeDEGtwocond), function(x) dataframeDEGtwocond[[x]] %>% dplyr::select(CombiDayDEG) %>% unlist() %>% as.character() %>% table() %>% as.data.frame() %>% top_n(1) %>% mutate("numberGene"=Freq/2) %>% as.data.frame())
maxGE2cond2<-lapply(names(dataframeDEGtwocond), function(x) dataframeDEGtwocond[[x]] %>% dplyr::select(CombiDaysUpDown) %>% unlist() %>% as.character() %>% table() %>% as.data.frame() %>% top_n(1) %>% mutate("numberGene"=Freq/2) %>% as.data.frame())
maxGE2cond<-lapply(1:length(maxGE2cond1), function(i) cbind(rbind(maxGE2cond1[[i]],maxGE2cond2[[i]]),numberDay=rep(2,dim(rbind(maxGE2cond1[[i]],maxGE2cond2[[i]]))[[1]]),typecell=rep(names(dataframeDEGtwocond )[[i]],dim(rbind(maxGE2cond1[[i]],maxGE2cond2[[i]]))[[1]])))
names(maxGE2cond) = names(dataframeDEGtwocond)

maxGE3cond1<-lapply(names(dataframeDEGthreecond), function(x) dataframeDEGthreecond[[x]] %>% dplyr::select(CombiDayDEG) %>% unlist() %>% as.character() %>% table() %>% as.data.frame() %>% top_n(1) %>% mutate("numberGene"=Freq/3) %>% as.data.frame())
maxGE3cond2<-lapply(names(dataframeDEGthreecond), function(x) dataframeDEGthreecond[[x]] %>% dplyr::select(CombiDaysUpDown) %>% unlist() %>% as.character() %>% table() %>% as.data.frame() %>% top_n(1) %>% mutate("numberGene"=Freq/3) %>% as.data.frame())
maxGE3cond<-lapply(1:length(maxGE3cond1), function(i) cbind(rbind(maxGE3cond1[[i]],maxGE3cond2[[i]]),numberDay=rep(3,dim(rbind(maxGE3cond1[[i]],maxGE3cond2[[i]]))[[1]]),typecell=rep(names(dataframeDEGthreecond )[[i]],dim(rbind(maxGE3cond1[[i]],maxGE3cond2[[i]]))[[1]])))
names(maxGE3cond) = names(dataframeDEGthreecond)

maxGE4cond1<-lapply(names(dataframeDEGfourcond), function(x) dataframeDEGfourcond[[x]] %>% dplyr::select(CombiDayDEG) %>% unlist() %>% as.character() %>% table() %>% as.data.frame() %>% top_n(1) %>% mutate("numberGene"=Freq/4) %>% as.data.frame())
maxGE4cond2<-lapply(names(dataframeDEGfourcond), function(x) dataframeDEGfourcond[[x]] %>% dplyr::select(CombiDaysUpDown) %>% unlist() %>% as.character() %>% table() %>% as.data.frame() %>% top_n(1) %>% mutate("numberGene"=Freq/4) %>% as.data.frame())
maxGE4cond<-lapply(1:length(maxGE4cond1), function(i) cbind(rbind(maxGE4cond1[[i]],maxGE4cond2[[i]]),numberDay=rep(4,dim(rbind(maxGE4cond1[[i]],maxGE4cond2[[i]]))[[1]]),typecell=rep(names(dataframeDEGfourcond)[[i]],dim(rbind(maxGE4cond1[[i]],maxGE4cond2[[i]]))[[1]])))
names(maxGE4cond) = names(dataframeDEGfourcond)

maxGEcond<-data.frame()
for(i in 1:length(names(maxGE1cond))){
  maxGEcond<-rbind(maxGEcond,maxGE1cond[[i]])
}
for(i in 1:length(names(maxGE2cond))){
  maxGEcond<-rbind(maxGEcond,maxGE2cond[[i]])
}
for(i in 1:length(names(maxGE3cond))){
  maxGEcond<-rbind(maxGEcond,maxGE3cond[[i]])
}
for(i in 1:length(names(maxGE4cond))){
  maxGEcond<-rbind(maxGEcond,maxGE4cond[[i]])
}
names(maxGEcond)<-c("PatternDaysDEG","Freq","numberGene","numberDay","typecell")
maxGEcond<- as.data.frame(maxGEcond) %>% dplyr::select(typecell,numberDay,PatternDaysDEG,numberGene) %>% arrange(typecell,numberDay,desc(numberGene))

```


`r kable(tablePercentgeneDEacrosscond,row.names = TRUE)%>% kable_paper("striped")`


```{r SankeyByTypeTotalprelude, fig.align='center' , echo=FALSE, warning=FALSE , message=FALSE}

###  CALCULATE
# Found the distribution of genes significatifs on Days Conditions by type cell
# Visualization with sankeyNetwork for genes DE on more 1 days
#
# ====================== Calculate =============================

#initialization variables for sankey diagram 
sourceNode<-c()
targetNode<-c()
valueEdge<-c()
Nbconnection<-c()
CombiDayDEG<-c()
for ( typeCell in Typecellv ) {
  tempoSigniDEgene<- signiDEgene %>% filter(type==typeCell)
  #Get list DEG by day
  uniqueIdByDay = lapply(unique(tempoSigniDEgene$day), function(x) filter(tempoSigniDEgene, day == x) %>% dplyr::select(id) %>% unlist() %>% as.character())
  names(uniqueIdByDay) = unique(tempoSigniDEgene$day)
  
  #Get all combination DEG 
  allcombinaison =ComplexHeatmap::make_comb_mat(uniqueIdByDay)
  #Extract pattern combination
  Sup2Combi<-allcombinaison[comb_degree(allcombinaison) >= 1]
  Node_Target<-names(comb_size(Sup2Combi))
  
  num_combi<-1
  
  #For one combination
  #Extract the number of day DEG
  #Found the name of combonation with day
  for ( name in Node_Target){
    particulName<-str_extract_all(name,boundary("character"))[[1]]
    names(particulName)<-1:length(particulName)
    
    Nodes<-lapply(1:length(particulName), function(i) if(particulName[i] == "1"){daysv[[typeCell]][i]} ) %>% unlist()
    if (str_count(name,"1")==1){CombiDayDEG<-c(CombiDayDEG,rep(str_c(Nodes, collapse = "."),4))} else {CombiDayDEG<-c(CombiDayDEG,rep(CombiDayDEG<-c(str_c(Nodes, collapse = ".")),str_count(name,"1")-1))}
    if (str_count(name,"1")==1){Nbconnection<-c(Nbconnection,rep(1,4))} else {Nbconnection<-c(Nbconnection,rep(str_count(name,"1"),str_count(name,"1")-1))}
    if (length(Nodes) == 1){
      sourceNode<-c(sourceNode,paste0(Nodes[1],"_",typeCell))
      targetNode<-c(targetNode,paste0(Nodes[1],"_",typeCell,"_only1"))
      valueEdge<-c(valueEdge,unname(comb_size(Sup2Combi)[num_combi]))
      sourceNode<-c(sourceNode,paste0(Nodes[1],"_",typeCell,"_only1"))
      targetNode<-c(targetNode,paste0(Nodes[1],"_",typeCell,"_only2"))
      valueEdge<-c(valueEdge,unname(comb_size(Sup2Combi)[num_combi]))
      sourceNode<-c(sourceNode,paste0(Nodes[1],"_",typeCell,"_only2"))
      targetNode<-c(targetNode,paste0(Nodes[1],"_",typeCell,"_only3"))
      valueEdge<-c(valueEdge,unname(comb_size(Sup2Combi)[num_combi]))
      sourceNode<-c(sourceNode,paste0(Nodes[1],"_",typeCell,"_only3"))
      targetNode<-c(targetNode,paste0(Nodes[1],"_",typeCell))
      valueEdge<-c(valueEdge,unname(comb_size(Sup2Combi)[num_combi]))
    }else{
      sourceNode<-c(sourceNode,paste0(Nodes[1],"_",typeCell))
      if(length(Nodes)-1 > 1){
        for (i in 2:(length(Nodes)-1)){
          sourceNode<-c(sourceNode,paste0(Nodes[i],"_",typeCell))
          targetNode<-c(targetNode,paste0(Nodes[i],"_",typeCell))
          valueEdge<-c(valueEdge,unname(comb_size(Sup2Combi)[num_combi]))
        }}
      targetNode<-c(targetNode,paste0(Nodes[length(Nodes)],"_",typeCell))
      valueEdge <-c(valueEdge,unname(comb_size(Sup2Combi)[num_combi]))
    }
    num_combi=num_combi+ 1
  } 
}  
#color
colorNbconnection<-c("white","#FABDA6","#D3DDDC","#C7B19C","#446455")
names(colorNbconnection)<-c("0","1","2","3","4")

colorCombiDayDEG<-c("#fef6be","#e2f4df","#dbeaf5","#d8d9ea",
                    "#d6efa6","#c2d9ed","#ffcce4","#a0dbb5","#d5b7d7","#8895c2",
                    "#008349","#065fa3","#d42857","#7e2175","#67081d")
names(colorCombiDayDEG)<-c("D0","D2","D4","D7"
                           ,"D0.D2","D0.D4","D0.D7","D2.D4","D2.D7","D4.D7",
                           "D0.D2.D4","D0.D2.D7","D0.D4.D7","D2.D4.D7","D0.D2.D4.D7")

# Make a connection data frame
# Total links
links <- data.frame(
  from=str_replace_all(sourceNode,"-",""),
  to=str_replace_all(targetNode,"-",""),
  Nbconnection=Nbconnection,
  CombiDayDEG=CombiDayDEG,
  substance= CombiDayDEG,
  quantity=sqrt(valueEdge)
)
links<-links %>% arrange(substance)
# From these flows we need to create a node data frame: it lists every entities involved in the flow
nodes <- data.frame(
  ID=c(as.character(links$from), as.character(links$to)) %>% 
    unique() %>% str_replace_all("-","")
)
nodes$label <- as.factor(c(as.character(links$from), as.character(links$to)) %>% 
                           unique())
nodes<-nodes  %>% mutate(day=sapply(strsplit(nodes$ID,"_"), `[`, 1)) %>% mutate(type=factor(sapply(strsplit(nodes$ID,"_"), `[`, 2),levels=str_replace_all(orderTypecell,"-",""),)) %>% mutate(loops=sapply(strsplit(nodes$ID,"_"), `[`, 3))
nbDay<-length(unique(nodes$day))
nbtype<-length(unique(nodes$type))
nodes<-nodes  %>% arrange(loops,type,day)
nodes<- nodes %>% mutate(label=lapply(1:length(nodes$day) ,function(i) ifelse( is.na(nodes[i,]$loops) == T,nodes[i,]$ID,"" )) %>% unlist()) 
nodes<- nodes %>%  mutate(ID=lapply( 1:length(nodes$day) ,function(i) ifelse( is.na(nodes[i,]$loops) == F,paste0(".",nodes[i,]$ID), nodes[i,]$ID)) %>% unlist())
poslabel=c("left","above","below","right")
nodes$label_pos<-c(rep("none",54),poslabel,poslabel,poslabel,"left","left","right","left","above","right")
xpos4<-c(-7,-2,3,8)
xpos3<-c(-2,3,8)
xpos2<-c(-2,3)
xpos1<-c(-2)
nodes$x<-c(rep(xpos4+2,3),xpos1+2,xpos2+2,xpos3+2,rep(xpos4,3),xpos1,xpos2,xpos3,rep(xpos4-2,3),xpos1-2,xpos2-2,xpos3-2,rep(xpos4,3),xpos1,xpos2,xpos3)
nodes<-nodes  %>% arrange(loops,day,type)
yposD2<-c(9,4,-2,-5,-8,-11)+1
yposD4<-c(9,3,-3,-8,-11)
yposD7<-c(9,4,-2,-13)+2
yposD0<-c(9,4,-2)+3
nodes$y<-c(yposD0+0.5,yposD2-0.5,yposD4-0.5,yposD7+0.5,yposD0+1,yposD2-1,yposD4-1,yposD7+1,yposD0+0.5,yposD2-0.5,yposD4-0.5,yposD7+0.5,yposD0,yposD2,yposD4,yposD7)
nodes$dir<-c(rep("up",3),rep("down",11),rep("up",4),rep("left",18),rep("down",3),rep("up",11),rep("down",4),rep("right",18))

dblue<-"#67001F"
# node style
ns <- list(type="arrow",gp=gpar(fill=dblue, col="white", lwd=3),
           length=0.5,
           label_gp=gpar(col="black", fontsize=6,fontface="bold"),
           mag_pos="label", mag_fmt="%.0f", mag_gp=gpar(fontsize=1,col="white"))

palettesNbconnection<-data.frame(substance=names(colorNbconnection),
                     color=colorNbconnection)
my_title_Nbconnection <- paste0("Total gene DEG group by type cell and number DEG on D0,D2,D4 and D7  \n Amount of gene (square root) depending on the combination of days on a cell type , padj<0.005 ")
attr(my_title_Nbconnection, "gp") <- grid::gpar(fontsize=12, fontface="bold", col="black")
links$substance<-as.character(links$Nbconnection)

palettesCombiDayDEG<-data.frame(substance=names(colorCombiDayDEG),
                     color=colorCombiDayDEG)
my_title_CombiDayDEG <- paste0("Total gene DEG group by type cell and CombiDayDEG  \n Amount of gene (square root) depending on the combination of days on a cell type , padj<0.005 ")
attr(my_title_CombiDayDEG, "gp") <- grid::gpar(fontsize=12, fontface="bold", col="black")

```



## III.1 Sankey plot - Show the days pattern of DEG {.tabset .tabset-fade .tabset-pills}
***

### Group by number of day DEG significant
```{r SankeyByTypeTotNbconnection, fig.align='center' , echo=FALSE, warning=FALSE , message=FALSE}

PantaRhei::sankey(nodes, links, palettesNbconnection,
                  max_width=0.05, rmin=0.5,
                  node_style=ns,
                  page_margin=c(0, 0.05, 0, 0.05),
                  title=my_title_Nbconnection, legend=gpar(fontsize=12, col="blue", ncols=1))

```

### Group by combination of days DEG significant

```{r SankeyByTypeTotaltest2, fig.align='center' , echo=FALSE, warning=FALSE , message=FALSE}
links$substance<-links$CombiDayDEG
PantaRhei::sankey(nodes, links, palettesCombiDayDEG,
                  max_width=0.05, rmin=0.5,
                  node_style=ns,
                  page_margin=c(0, 0.05, 0, 0.05),
                  title=my_title_CombiDayDEG, legend=gpar(fontsize=12, col="blue", ncols=1))
```

```{r SankeyByTypefunction, fig.align='center' , echo=FALSE, warning=FALSE , message=FALSE}

#tableSankeyplot function to extrat table nodes sources and target with value , for a sankey plot
#plot gathered gene with the same pattern (for one type cell, witch days are DEG and in witch sens) and links days in function of sens of log2foldchange
#so if gene is UP, it is more exessed in OLD condition than Young
#signiDEgene = table with all DEG padj<0.05
#vectTypeCell = type cell selected, vector possible
#minCombi = founds genes with at least 1 or 2 days DEG
#NODIFF = Nodes inclus days UNDIFF

tableSankeyplot<- function(signiDEgene,vectTypeCell,minCombi,NODIFF){
sourceNode<-c()
targetNode<-c()
valueEdge<-c()
NbUpDown<-c()
Nbconnection<-c()
CombiDayDEG<-c()

for ( typeCell in vectTypeCell ) {
  #prepare list gene_SENS DEG by day
  tempoSigniDEgene<- signiDEgene %>% filter(type==typeCell)
  uniqueIdByDayUP = lapply(unique(tempoSigniDEgene$day), function(x) filter(tempoSigniDEgene, day == x)  %>% filter(log2FoldChange > 0) %>% dplyr::select(id) %>% unlist() %>% as.character())
  names(uniqueIdByDayUP) = unique(paste0(tempoSigniDEgene$day,"_UP"))
  uniqueIdByDayDOWN = lapply(unique(tempoSigniDEgene$day), function(x) filter(tempoSigniDEgene, day == x)  %>% filter(log2FoldChange < 0) %>% dplyr::select(id) %>% unlist() %>% as.character())
  names(uniqueIdByDayDOWN) = unique(paste0(tempoSigniDEgene$day,"_DOWN"))
  uniqueIdByDay<-c(uniqueIdByDayUP,uniqueIdByDayDOWN)
  
  #Get combinaison
  allcombinaison =ComplexHeatmap::make_comb_mat(uniqueIdByDay)
  Sup2Combi<-allcombinaison[comb_degree(allcombinaison) >= minCombi]
  
  #Get name combinaison
  Node_Target<-names(comb_size(Sup2Combi))
  if(length(Node_Target) < minCombi){next}
  num_combi<-1
  
  #For one combination
  #Extract the number of day DEG
  #Found the name of combination with day+sens(UP,DOWN)
  #Extract Day + sens (UP, DOWN,UNDIFF)
  for ( name in Node_Target){
    particulName<-str_extract_all(name,boundary("character"))[[1]]
    names(particulName)<-1:length(particulName)
    #extract day DEG UP
    tempoName1<-lapply(1:(length(particulName)/2), function(i) if(particulName[[i]] == "1"){paste0(daysv[[typeCell]][i],"_UP")} ) %>% unlist()
    #extract day DEG DOWN
    tempoName2<-lapply(((length(particulName)/2)+1):length(particulName), function(i) if(particulName[[i]] == "1"){paste0(daysv[[typeCell]][i-(length(particulName)/2)],"_DOWN")} ) %>% unlist()
    #Extract day UNDIFF
    tempoName3<-lapply(1:(length(particulName)/2), function(i) if(particulName[[i]] == "0" && particulName[[i+(length(particulName)/2)]] == "0"){paste0(daysv[[typeCell]][i],"_UNDIFF")} ) %>% unlist()
    
    #Extract only day DEG
    tempoName1bis<-lapply(1:(length(particulName)/2), function(i) if(particulName[[i]] == "1"){daysv[[typeCell]][i]} ) %>% unlist()
    tempoName2bis<-lapply(((length(particulName)/2)+1):length(particulName), function(i) if(particulName[[i]] == "1"){daysv[[typeCell]][i-(length(particulName)/2)]} ) %>% unlist()
    tempoName<-str_c(sort(c(tempoName1bis,tempoName2bis)),collapse = ".")
    
    if (NODIFF == T ){
    Nodes<-sort(c(tempoName1,tempoName2,tempoName3))
    CombiDayDEG<-c(CombiDayDEG,rep(tempoName,length(Nodes)-1))
    Nbconnection<-c(Nbconnection,rep(length(c(tempoName1,tempoName2)),length(Nodes)-1))
    NbUpDown<-c(NbUpDown,rep(paste0(ifelse(length(tempoName1)==0,"",paste0(length(tempoName1),"_UP.")),ifelse(length(tempoName2)==0,"",paste0(length(tempoName2),"_DOWN"))),length(Nodes)-1))
    
    }else{    
    Nodes<-sort(c(tempoName1,tempoName2))
    if (length(Nodes)==1){Nbconnection<-c(Nbconnection,rep(length(c(tempoName1,tempoName2)),4))} else {Nbconnection<-c(Nbconnection,rep(length(c(tempoName1,tempoName2)),length(Nodes)-1))}
    if (length(Nodes)==1){CombiDayDEG<-c(CombiDayDEG,rep(tempoName,4))} else {CombiDayDEG<-c(CombiDayDEG,rep(tempoName,length(Nodes)-1))}
    if (length(Nodes)==1){NbUpDown<-c(NbUpDown,rep(paste0(ifelse(length(tempoName1)==0,"",paste0(length(tempoName1),"_UP.")),ifelse(length(tempoName2)==0,"",paste0(length(tempoName2),"_DOWN"))),4))} else {NbUpDown<-c(NbUpDown,rep(paste0(ifelse(length(tempoName1)==0,"",paste0(length(tempoName1),"_UP.")),ifelse(length(tempoName2)==0,"",paste0(length(tempoName2),"_DOWN"))),length(Nodes)-1))}
    
    }

    #Fill source node, target node (day consecutive) + nb gene with this combination
    if (length(Nodes) == 1){
        sourceNode<-c(sourceNode,Nodes[1])
        targetNode<-c(targetNode,paste0(Nodes[1],"_only1"))
        valueEdge<-c(valueEdge,unname(comb_size(Sup2Combi)[num_combi]))
        sourceNode<-c(sourceNode,paste0(Nodes[1],"_only1"))
        targetNode<-c(targetNode,paste0(Nodes[1],"_only2"))
        valueEdge<-c(valueEdge,unname(comb_size(Sup2Combi)[num_combi]))
        sourceNode<-c(sourceNode,paste0(Nodes[1],"_only2"))
        targetNode<-c(targetNode,paste0(Nodes[1],"_only3"))
        valueEdge<-c(valueEdge,unname(comb_size(Sup2Combi)[num_combi]))
        sourceNode<-c(sourceNode,paste0(Nodes[1],"_only3"))
        targetNode<-c(targetNode,Nodes[1])
        valueEdge<-c(valueEdge,unname(comb_size(Sup2Combi)[num_combi]))
    }else{
    sourceNode<-c(sourceNode,Nodes[1])
    if(length(Nodes)-1 > 1){
      for (i in 2:(length(Nodes)-1)){
        sourceNode<-c(sourceNode,Nodes[i])
        targetNode<-c(targetNode,Nodes[i])
        valueEdge<-c(valueEdge,unname(comb_size(Sup2Combi)[num_combi]))
      }}
    targetNode<-c(targetNode,Nodes[length(Nodes)])
    valueEdge <-c(valueEdge,unname(comb_size(Sup2Combi)[num_combi]))
    }
    
    num_combi=num_combi+ 1
  }

}

links <- data.frame(
  from=str_replace_all(sourceNode,"-",""),
  to=str_replace_all(targetNode,"-",""),
  NbUpDown= NbUpDown,
  Nbconnection=Nbconnection,
  CombiDayDEG=CombiDayDEG,
  quantity=valueEdge
)


return(links)
}

# sankeyPantaRhei , made a sankey plot, where you can choice coordinate of nodes, label nodes,group color links by group (NbUpDown,Nbconnection,CombiDayDEG)
#links1 = output of tableSankeyplot with at least nodes from, nodes to, value of the flow, and group
#dblue= color of nodes
#group=
#NbUpDown: group links by pattern number of UP and number od DOWN
#Nbconnection: group by number of day DEG
#CombiDayDEG: group by pattern of days DEG
#colorvect= vector color with names unique(links1$group)
#outputprefix= start of plot title
#sqrt= T/F, value in square root
#NODIFF= T/F , nodes UNDIFF?
sankeyPantaRhei<-function(links1,dblue,group,colorvect,outputprefix,sqrt,NODIFF){
  if ( sqrt == T){
    links1$quantity<-sqrt(links1$quantity)
    titlesqrt<-"(sqrt)"}else{titlesqrt<-""
    }
  links <- data.frame(
    from=links1$from,
    to=links1$to,
    substance= as.factor(links1 %>% dplyr::select(group) %>% unlist()),
    #substance= as.factor(links1$CombiDayDEG),
    quantity=links1$quantity
  )
  links<-links %>% arrange(substance)
  # From these flows we need to create a node data frame: it lists every entities involved in the flow
  nodes <- data.frame(
    ID=c(as.character(links$from), as.character(links$to)) %>% 
      unique() %>% str_replace_all("-","")
  )
  nodes$label <- as.factor(c(as.character(links$from), as.character(links$to)) %>% 
                             unique())
  nodes<-nodes  %>% mutate(day=sapply(strsplit(nodes$ID,"_"), `[`, 1)) %>% mutate(sens=sapply(strsplit(nodes$ID,"_"), `[`, 2)) %>% mutate(loops=sapply(strsplit(nodes$ID,"_"), `[`, 3))
  nbDay<-length(unique(nodes$day))
  nodes<-nodes  %>% arrange(loops,sens,day)
  if("only1" %in% nodes$loops){only=T}else{only=F}
  if(length(nodes %>% filter(sens== "UNDIFF") %>% dplyr::select(sens) %>% unlist()) >=1){
    replabel=3
    titleNODIFF="UNDIFF"}else{
    NODIFF = F
    replabel=2
    titleNODIFF=""}
  if (nbDay == 4){
    poslabel=c("left","above","below","right")
    xpos<-c(-7,-2,3,8)
    ypos<-c(-2,-3,-4,-5)
  } else if(nbDay == 3){
    poslabel=c("left","above","right")
    xpos<-c(-7,-2,3)
    ypos<-c(-2,-3,-4)
  } else{
      poslabel=c("left","right")
      xpos<-c(-7,3)
      ypos<-c(-2,-3)
  }

  nodes<- nodes %>% mutate(label=lapply(1:length(nodes$sens) ,function(i) ifelse( is.na(nodes[i,]$loops) == T,nodes[i,]$ID,"" )) %>% unlist()) 
  nodes<- nodes %>%  mutate(ID=lapply( 1:length(nodes$sens) ,function(i) ifelse( is.na(nodes[i,]$loops) == F,paste0(".",nodes[i,]$ID), nodes[i,]$ID)) %>% unlist())
  if(only == T){nodes$label_pos<-c(rep("none",nbDay*6),rep(poslabel,replabel))}else{nodes$label_pos<-c(rep(poslabel,replabel))}
  nodes$label_align<-rep("",length(nodes$ID))
  if(NODIFF == T){nodes$x<-c(xpos,xpos+1,xpos)}else{ if(only == T){nodes$x<-c(xpos+2,xpos+2,xpos,xpos,xpos-2,xpos-2,rep(xpos,2))} else{nodes$x<-c(xpos,xpos)}}
  if(NODIFF == T){nodes$y<-c(ypos,rep(0,nbDay),ypos*-1)}else{ if(only == T){nodes$y<-c(ypos-1,(ypos*-1)+1,ypos-2,(ypos*-1)+2,ypos-1,(ypos*-1)+1,ypos,ypos*-1)}else{nodes$y<-c(ypos,ypos*-1)}}
  if(only == T){nodes$dir<-c(rep("down",nbDay),rep("up",nbDay),rep("left",nbDay*2),rep("up",nbDay),rep("down",nbDay),rep("right",nbDay*2))}else{nodes$dir<-rep("right",nbDay*replabel)}
  
  palettes<-data.frame(substance=names(colorvect),
                       color=colorvect)
  my_title <- paste0(outputprefix," group by ",group," with sens UP DOWN ",titleNODIFF,"\n Amount of gene ",titlesqrt," depending on the combination of days on a cell type , padj<0.005 ")
  attr(my_title, "gp") <- grid::gpar(fontsize=12, fontface="bold", col="black")
  
  # node style
  ns <- list(type="arrow",gp=gpar(fill=dblue, col="white", lwd=3),
             length=0.5,
             label_gp=gpar(col="black", fontsize=6,fontface="bold"),
             mag_pos="label", mag_fmt="%.0f", mag_gp=gpar(fontsize=6,fontface="bold",col=dblue))
  
  #pdf(paste0(analyse,"test_group_",group,"_",titleNODIFF,'_',sqrt,'_',outputprefix,"_sankeyColorUPDOWN.pdf"), width=10, height=7) # Set up PDF device
  PantaRhei::sankey(nodes, links, palettes,
         max_width=0.15, rmin=0.5,
         node_style=ns,
         page_margin=c(0, 0.05, 0, 0.05),
         title=my_title, legend=gpar(fontsize=8, col="blue", ncols=1))
  #dev.off()
  
}

#Vectcorlor in function group chosen
colorNbUpDown<-c("plum","plum4",
                 "mediumpurple1","mediumpurple4",
                 "violetred1","violetred4",
                 "LightCyan","Aquamarine","skyblue2","royalblue3",
                 "LightPink","indianred1","firebrick1","red3" )
#color for sankeyNetwork
colorNbUpDown2<-c("#D8BFD8","#DA70D6",
                  "#9370DB","#663399",
                  "#FF69B4","#C71585",
                  "#E0FFFF","#7FFFD4","#6495ED","#00008B",
                  "#FFB6C1","#F08080","#DC143C","#8B0000" )
names(colorNbUpDown)<-c("1_UP.1_DOWN" ,"2_UP.2_DOWN",
                        "1_UP.2_DOWN", "1_UP.3_DOWN" ,
                        "2_UP.1_DOWN","3_UP.1_DOWN",
                        "1_DOWN","2_DOWN","3_DOWN" ,"4_DOWN",
                        "1_UP.","2_UP.","3_UP.","4_UP.")
colorNbconnection<-c("white","#FABDA6","#D3DDDC","#C7B19C","#446455")
names(colorNbconnection)<-c("0","1","2","3","4")

colorCombiDayDEG<-c("#fef6be","#e2f4df","#dbeaf5","#d8d9ea",
                    "#d6efa6","#c2d9ed","#ffcce4","#a0dbb5","#d5b7d7","#8895c2",
                    "#008349","#065fa3","#d42857","#7e2175","#67081d")
names(colorCombiDayDEG)<-c("D0","D2","D4","D7"
                           ,"D0.D2","D0.D4","D0.D7","D2.D4","D2.D7","D4.D7",
                           "D0.D2.D4","D0.D2.D7","D0.D4.D7","D2.D4.D7","D0.D2.D4.D7")
dataFrameTotalUPDOWN<-readRDS(paste0(pathcombinaisontableUPDOWN,'.rds'))

```


## III.2 Sankey plot - Show the days pattern of DEG UP and DOWN significant

### ECs{.tabset .tabset-fade .tabset-pills}
***

#### Group by combination of number of days DEG UP and DOWN significant

```{r SankeyECsnbUpDown, fig.align='center' , echo=FALSE, warning=FALSE , message=FALSE}
typeCell="ECs"
tableSankeyplotFAPs<-tableSankeyplot(signiDEgene,typeCell,1,F)
sankeyPantaRhei(tableSankeyplotFAPs,colorsType[[typeCell]],"NbUpDown",colorNbUpDown,paste0(typeCell,"_combisupp1"),T,F)
```

#### Group by number of day DEG UP and DOWN significant

```{r SankeyECspatternDayUpDown, fig.align='center' , echo=FALSE, warning=FALSE , message=FALSE}
typeCell="ECs"
tableSankeyplotFAPs<-tableSankeyplot(signiDEgene,typeCell,1,F)
sankeyPantaRhei(tableSankeyplotFAPs,colorsType[[typeCell]],"CombiDayDEG",colorCombiDayDEG,paste0(typeCell,"_combisupp1"),T,F)
```

#### Group by combination of days DEG UP and DOWN significant

```{r SankeyECsnbDEG, fig.align='center' , echo=FALSE, warning=FALSE , message=FALSE}
typeCell="ECs"
tableSankeyplotFAPs<-tableSankeyplot(signiDEgene,typeCell,1,F)
sankeyPantaRhei(tableSankeyplotFAPs,colorsType[[typeCell]],"Nbconnection",colorNbconnection,paste0(typeCell,"_combisupp1"),T,F)
```

#### Table with pattern the most present 

```{r tablECs, fig.align='center' , echo=FALSE, warning=FALSE , message=FALSE}
maxGEcondtempo<- maxGEcond %>% filter(typecell==typeCell)
maxGEcondtempo
```

### {-}

```{r tableECsUPDOWN, echo=FALSE, warning=FALSE}
tablefilter<-dataFrameTotalUPDOWN %>% filter(type==typeCell) %>% arrange(nbDayDE,CombiDaysUpDown,symbol,padj) %>% select(nbDayDE,CombiDaysUpDown,symbol,id,day,type,CombiDayDEG,NbUpDown,padj,log2FoldChange,meanCountNormalizedYoung,meanCountNormalizedOld)
DT::datatable(tablefilter, filter = 'top', extensions = c('RowGroup','Select', 'SearchPanes'),options = list(rowGroup = list(dataSrc = c(1,2)),dom = 'Pfrtip',columnDefs = list(list(searchPanes = list(show = FALSE), targets = 4:11 ))),selection = 'none',escape = F,rownames=F)
```

### FAPs{.tabset .tabset-fade .tabset-pills}
***

#### Group by combination of number of days DEG UP and DOWN significant

```{r SankeyFAPsnbUpDown, fig.align='center' , echo=FALSE, warning=FALSE , message=FALSE}
typeCell="FAPs"
tableSankeyplotFAPs<-tableSankeyplot(signiDEgene,typeCell,1,F)
sankeyPantaRhei(tableSankeyplotFAPs,colorsType[[typeCell]],"NbUpDown",colorNbUpDown,paste0(typeCell,"_combisupp1"),T,F)
```

#### Group by number of day DEG UP and DOWN significant

```{r SankeyFAPspatternDayUpDown, fig.align='center' , echo=FALSE, warning=FALSE , message=FALSE}
typeCell="FAPs"
tableSankeyplotFAPs<-tableSankeyplot(signiDEgene,typeCell,1,F)
sankeyPantaRhei(tableSankeyplotFAPs,colorsType[[typeCell]],"CombiDayDEG",colorCombiDayDEG,paste0(typeCell,"_combisupp1"),T,F)
```

#### Group by combination of days DEG UP and DOWN significant

```{r SankeyFAPsnbDEG, fig.align='center' , echo=FALSE, warning=FALSE , message=FALSE}
typeCell="FAPs"
tableSankeyplotFAPs<-tableSankeyplot(signiDEgene,typeCell,1,F)
sankeyPantaRhei(tableSankeyplotFAPs,colorsType[[typeCell]],"Nbconnection",colorNbconnection,paste0(typeCell,"_combisupp1"),T,F)
```

#### Table with pattern the most present 

```{r tablFAPs, fig.align='center' , echo=FALSE, warning=FALSE , message=FALSE}
maxGEcondtempo<- maxGEcond %>% filter(typecell==typeCell)
maxGEcondtempo
```

### {-}

```{r tableFAPsUPDOWN, echo=FALSE, warning=FALSE}
tablefilter<-dataFrameTotalUPDOWN %>% filter(type==typeCell) %>% arrange(nbDayDE,CombiDaysUpDown,symbol,padj) %>% select(nbDayDE,CombiDaysUpDown,symbol,id,day,type,CombiDayDEG,NbUpDown,padj,log2FoldChange,meanCountNormalizedYoung,meanCountNormalizedOld)
DT::datatable(tablefilter, filter = 'top', extensions = c('RowGroup','Select', 'SearchPanes'),options = list(rowGroup = list(dataSrc = c(1,2)),dom = 'Pfrtip',columnDefs = list(list(searchPanes = list(show = FALSE), targets = 4:11 ))),selection = 'none',escape = F,rownames=F)
```

### MuSCs{.tabset .tabset-fade .tabset-pills}
***

#### Group by combination of number of days DEG UP and DOWN significant

```{r SankeyMuSCsnbUpDown, fig.align='center' , echo=FALSE, warning=FALSE , message=FALSE}
typeCell="MuSCs"
tableSankeyplotFAPs<-tableSankeyplot(signiDEgene,typeCell,1,F)
sankeyPantaRhei(tableSankeyplotFAPs,colorsType[[typeCell]],"NbUpDown",colorNbUpDown,paste0(typeCell,"_combisupp1"),T,F)
```

#### Group by number of day DEG UP and DOWN significant

```{r SankeyMuSCspatternDayUpDown, fig.align='center' , echo=FALSE, warning=FALSE , message=FALSE}
typeCell="MuSCs"
tableSankeyplotFAPs<-tableSankeyplot(signiDEgene,typeCell,1,F)
sankeyPantaRhei(tableSankeyplotFAPs,colorsType[[typeCell]],"CombiDayDEG",colorCombiDayDEG,paste0(typeCell,"_combisupp1"),T,F)
```

#### Group by combination of days DEG UP and DOWN significant

```{r SankeyMuSCsnbDEG, fig.align='center' , echo=FALSE, warning=FALSE , message=FALSE}
typeCell="MuSCs"
tableSankeyplotFAPs<-tableSankeyplot(signiDEgene,typeCell,1,F)
sankeyPantaRhei(tableSankeyplotFAPs,colorsType[[typeCell]],"Nbconnection",colorNbconnection,paste0(typeCell,"_combisupp1"),T,F)
```

#### Table with pattern the most present 

```{r tablMusc, fig.align='center' , echo=FALSE, warning=FALSE , message=FALSE}
maxGEcondtempo<- maxGEcond %>% filter(typecell==typeCell)
maxGEcondtempo
```

### {-}

```{r tableMuSCsUPDOWN, echo=FALSE, warning=FALSE}
tablefilter<-dataFrameTotalUPDOWN %>% filter(type==typeCell) %>% arrange(nbDayDE,CombiDaysUpDown,symbol,padj) %>% select(nbDayDE,CombiDaysUpDown,symbol,id,day,type,CombiDayDEG,NbUpDown,padj,log2FoldChange,meanCountNormalizedYoung,meanCountNormalizedOld)
DT::datatable(tablefilter, filter = 'top', extensions = c('RowGroup','Select', 'SearchPanes'),options = list(rowGroup = list(dataSrc = c(1,2)),dom = 'Pfrtip',columnDefs = list(list(searchPanes = list(show = FALSE), targets = 4:11 ))),selection = 'none',escape = F,rownames=F)
```

### Inflammatory-Mac{.tabset .tabset-fade .tabset-pills}
***

#### Group by combination of number of days DEG UP and DOWN significant

```{r SankeyInflammatorysnbUpDown, fig.align='center' , echo=FALSE, warning=FALSE , message=FALSE}
typecell="Inflammatory-Mac"
tableSankeyplotFAPs<-tableSankeyplot(signiDEgene,typecell,1,F)
sankeyPantaRhei(tableSankeyplotFAPs,colorsType[[typecell]],"NbUpDown",colorNbUpDown,paste0(typecell,"_combisupp1"),T,F)
```

#### Group by number of day DEG UP and DOWN significant

```{r SankeyInflammatorypatternDayUpDown, fig.align='center' , echo=FALSE, warning=FALSE , message=FALSE}
typecell="Inflammatory-Mac"
tableSankeyplotFAPs<-tableSankeyplot(signiDEgene,typecell,1,F)
sankeyPantaRhei(tableSankeyplotFAPs,colorsType[[typecell]],"CombiDayDEG",colorCombiDayDEG,paste0(typecell,"_combisupp1"),T,F)
```

#### Group by combination of days DEG UP and DOWN significant

```{r SankeyInflammatorynbDEG, fig.align='center' , echo=FALSE, warning=FALSE , message=FALSE}
typecell="Inflammatory-Mac"
tableSankeyplotFAPs<-tableSankeyplot(signiDEgene,typecell,1,F)
sankeyPantaRhei(tableSankeyplotFAPs,colorsType[[typecell]],"Nbconnection",colorNbconnection,paste0(typecell,"_combisupp1"),T,F)
```

#### Table with pattern the most present 

```{r tablinflammatory, fig.align='center' , echo=FALSE, warning=FALSE , message=FALSE}
maxGEcondtempo<- maxGEcond %>% filter(typecell==typecell)
maxGEcondtempo
```

### {-}

```{r tableInflammatoryUPDOWN, echo=FALSE, warning=FALSE}
tablefilter<-dataFrameTotalUPDOWN %>% filter(type==typecell) %>% arrange(nbDayDE,CombiDaysUpDown,symbol,padj) %>% select(nbDayDE,CombiDaysUpDown,symbol,id,day,type,CombiDayDEG,NbUpDown,padj,log2FoldChange,meanCountNormalizedYoung,meanCountNormalizedOld)
DT::datatable(tablefilter, filter = 'top', extensions = c('RowGroup','Select', 'SearchPanes'),options = list(rowGroup = list(dataSrc = c(1,2)),dom = 'Pfrtip',columnDefs = list(list(searchPanes = list(show = FALSE), targets = 4:11 ))),selection = 'none',escape = F,rownames=F)
```

### Resolving-Mac{.tabset .tabset-fade .tabset-pills}
***

#### Group by combination of number of days DEG UP and DOWN significant

```{r SankeyResolvingnbUpDown, fig.align='center' , echo=FALSE, warning=FALSE , message=FALSE}
typeCell="Resolving-Mac"
tableSankeyplotFAPs<-tableSankeyplot(signiDEgene,typeCell,1,F)
sankeyPantaRhei(tableSankeyplotFAPs,colorsType[[typeCell]],"NbUpDown",colorNbUpDown,paste0(typeCell,"_combisupp1"),T,F)
```

#### Group by number of day DEG UP and DOWN significant

```{r SankeyResolvingpatternDayUpDown, fig.align='center' , echo=FALSE, warning=FALSE , message=FALSE}
typeCell="Resolving-Mac"
tableSankeyplotFAPs<-tableSankeyplot(signiDEgene,typeCell,1,F)
sankeyPantaRhei(tableSankeyplotFAPs,colorsType[[typeCell]],"CombiDayDEG",colorCombiDayDEG,paste0(typeCell,"_combisupp1"),T,F)
```

#### Group by combination of days DEG UP and DOWN significant

```{r SankeyResolvingnbDEG, fig.align='center' , echo=FALSE, warning=FALSE , message=FALSE}
typeCell="Resolving-Mac"
tableSankeyplotFAPs<-tableSankeyplot(signiDEgene,typeCell,1,F)
sankeyPantaRhei(tableSankeyplotFAPs,colorsType[[typeCell]],"Nbconnection",colorNbconnection,paste0(typeCell,"_combisupp1"),T,F)
```

#### Table with pattern the most present 

```{r tablresolving, fig.align='center' , echo=FALSE, warning=FALSE , message=FALSE}
maxGEcondtempo<- maxGEcond %>% filter(typecell==typeCell)
maxGEcondtempo
```

### {-}

```{r tableResolvingUPDOWN, echo=FALSE, warning=FALSE}
tablefilter<-dataFrameTotalUPDOWN %>% filter(type==typeCell) %>% arrange(nbDayDE,CombiDaysUpDown,symbol,padj) %>% select(nbDayDE,CombiDaysUpDown,symbol,id,day,type,CombiDayDEG,NbUpDown,padj,log2FoldChange,meanCountNormalizedYoung,meanCountNormalizedOld)
DT::datatable(tablefilter, filter = 'top', extensions = c('RowGroup','Select', 'SearchPanes'),options = list(rowGroup = list(dataSrc = c(1,2)),dom = 'Pfrtip',columnDefs = list(list(searchPanes = list(show = FALSE), targets = 4:11 ))),selection = 'none',escape = F,rownames=F)
```

# IV. Total table gene differential express significant

[linksTableUpDown](./tableDEGsignificantUpDown.html)

# V. Next report on genes set enrichment

[linksreportgsea](./fgseaReport.html)
